packages/patternfly-4/react-charts/src/components/ChartDonut/ChartDonut.js

```diff
@@ -1,21 +1,130 @@
    */
-  '': PropTypes.any
+  '': PropTypes.any,
+  /**
+   * The height props specifies the height the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  height: PropTypes.number,
+  /**
+   * The subtitle for the donut chart
+   */
+  subTitle: PropTypes.string,
+  /*
+   * Specifies the theme color; blue (default), green, or multi-color. Overridden by the theme property.
+   */
+  themeColor: PropTypes.string,
+  /*
+   * Specifies the theme variant; 'dark' or 'light' (default). Overridden by the theme property.
+   */
+  themeVariant: PropTypes.string,
+  /**
+   * The title for the donut chart
+   */
+  title: PropTypes.string,
+  /**
+   * The width props specifies the width of the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number
 };
 
+class ChartDonut extends React.Component {
+  childeRef = React.createRef();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      height: this.props.height || 0,
+      width: this.props.width || 0
+    };
+  }
+
+  componentDidMount() {
+    // Skip resize handler if height and width were provided
+    if (!(this.state.height || this.state.width)) {
+      setTimeout(() => {
+        window.addEventListener('resize', this.handleResize);
+      });
+      this.handleResize();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  handleResize = () => {
+    if (this.childeRef.current && this.childeRef.current.parentNode) {
+      this.setState({
+        height: this.childeRef.current.parentNode.clientHeight,
+        width: this.childeRef.current.parentNode.clientWidth
+      });
+    }
+  };
+
+  render() {
+    const { subTitle, theme, themeColor, themeVariant, title, ...rest } = this.props;
+    const { height, width } = this.state;
+
+    const chartTheme = theme || getDonutTheme(themeColor, themeVariant);
+    const showLabels = subTitle || title;
+    const innerRadius = (width - 32) / 2;
+
+    if (showLabels && this.props.standalone !== false) {
+      return (
+        <div ref={this.childeRef}>
```

Comment:
We shouldn't need to wrap this in a div. If the standalone prop is `true`, we should wrap the `ChartLabel` and `VictoryPie` in the `ChartContainer`. If standalone is `false`, we should wrap them in a React fragment so that only the inner svg graphics are returned (the user is responsible for providing the svg wrapper).

Reply:
Wrapping in a div to get the parent container's height and width. If standalone={false} there is no div

Reply:
Something like this:
```jsx
const chart = (
      <React.Fragment>
        <ChartLabel
          style={[styles.label.title, styles.label.subTitle]}
          text={[title, subTitle]}
          textAnchor="middle"
          verticalAnchor="middle"
          x={width / 2}
          y={height / 2}
        />
        <VictoryPie
          innerRadius={innerRadius}
          labelComponent={<ChartTooltip theme={theme} />}
          standalone={false}
          theme={chartTheme}
          {...rest}
        />
      </React.Fragment>
    );
    return standalone ? <ChartContainer>{chart}</ChartContainer> : chart;
```

Reply:
This would match the behavior of setting `standalone === false` on a primitive Victory chart.

Reply:
I don't believe we should set a default height and width here. This would behave differently than the responsive containers which are 100% height and width.

Reply:
From the Victory docs:
> VictoryContainer provides a top-level <svg> element for other Victory components to render within. **By default, VictoryContainer renders responsive SVGs.** VictoryContainer also provides a Portal container that can be accessed via VictoryPortal in order to render specified children above others.

Reply:
Victory docs on `height` and `width` prop of container components:
> The height prop determines the height of the containing <svg>. By default VictoryContainer renders responsive containers with the viewBox attribute set to viewBox="0, 0, width, height" and width="100%", height="100%". In responsive containers, the width and height props affect the aspect ratio of the rendered component, while the absolute width and height are determined by the container. To render a static container, set responsive={false}

> The width prop determines the width of the containing <svg>. By default VictoryContainer renders responsive containers with the viewBox attribute set to viewBox="0, 0, width, height" and width="100%", height="auto". In responsive containers, the width and height props affect the aspect ratio of the rendered component, while the absolute width and height are determined by the container. To render a static container, set responsive={false}

Reply:
I read those docs, too. :)

My issue is that we're setting a default height and width inside the component. The Victory greyscale theme applies different values for height and width.

Maybe if we can pull those values from the theme itself? That way we don't end up with different aspect ratios -- I'll try that instead.

Reply:
Oh, I see, so you mean the height and width props of the `VictoryPie` component itself, not contianer?